File: src/test/java/uk/co/eelpieconsulting/feedlistener/rss/FeedParserTest.java
package uk.co.eelpieconsulting.feedlistener.rss;

import com.github.kittinunf.result.Result;
import com.rometools.rome.feed.synd.SyndEntry;
import com.rometools.rome.feed.synd.SyndFeed;
import org.apache.commons.io.IOUtils;
import org.junit.jupiter.api.Test;

import java.io.FileInputStream;
import java.util.Iterator;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class FeedParserTest {

    private final FeedParser feedParser = new FeedParser();

    @Test
    public void canParseFeedBytesIntoSyndFeed() throws Exception {
        String input = IOUtils.toString(new FileInputStream(this.getClass().getClassLoader().getResource("wcc-news.xml").getFile()));
        System.out.println(input);

        Result<SyndFeed, Exception> result = feedParser.parseSyndFeed(input.getBytes());

        assertEquals("Wellington City Council - News", result.get().getTitle());
    }

    @Test
    public void needToStripNBSPFromMalformedFeeds() throws Exception {
        String input = IOUtils.toString(new FileInputStream(this.getClass().getClassLoader().getResource("vinnies-news.xml").getFile()));

        Result<SyndFeed, Exception> result = feedParser.parseSyndFeed(input.getBytes());

        assertEquals("Latest News - St Vincent de Paul Society Wellington", result.get().getTitle());
    }

    @Test
    public void whatsUpWithCricketWellingtonsFeed() throws Exception {
        String input = IOUtils.toString(new FileInputStream(this.getClass().getClassLoader().getResource("cricketwellington.xml").getFile()));

        Result<SyndFeed, Exception> result = feedParser.parseSyndFeed(input.getBytes());

        assertEquals("Cricket Wellington", result.get().getTitle());
        assertEquals(50, result.get().getEntries().size());


        final Iterator<SyndEntry> feedItemsIterator = result.get().getEntries().iterator();
        SyndEntry next = feedItemsIterator.next();

        assertEquals("Big names return for Firebirds", next.getTitle());
        assertEquals("21 Feb 2020 11:46:19 GMT", next.getPublishedDate().toGMTString());
    }

}

File: src/test/kotlin/uk/co/eelpieconsulting/feedlistener/rss/FeedParserTest.kt
package uk.co.eelpieconsulting.feedlistener.rss

import com.rometools.rome.feed.synd.SyndEntry
import org.apache.commons.io.IOUtils
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import java.io.FileInputStream
import java.nio.charset.StandardCharsets

class FeedParserTest {
    private val feedParser: FeedParser = FeedParser()
    
    @Test
    fun canParseFeedBytesIntoSyndFeed() {
        val input = readTestFile("wcc-news.xml")

        val result= feedParser.parseSyndFeed(input.toByteArray())

        assertEquals("Wellington City Council - News", result.get().title)
    }

    @Test
    fun needToStripNBSPFromMalformedFeeds() {
        val input = readTestFile("vinnies-news.xml")

        val result= feedParser.parseSyndFeed(input.toByteArray())

        assertEquals("Latest News - St Vincent de Paul Society Wellington", result.get().title)
    }

    @Test
    fun whatsUpWithCricketWellingtonsFeed() {
        val input = readTestFile("cricketwellington.xml")

        val result= feedParser.parseSyndFeed(input.toByteArray())

        assertEquals("Cricket Wellington", result.get().title)
        assertEquals(50, result.get().entries.size)
        val feedItemsIterator: Iterator<SyndEntry> = result.get().entries.iterator()
        val next: SyndEntry = feedItemsIterator.next()
        assertEquals("Big names return for Firebirds", next.title)
        assertEquals("21 Feb 2020 11:46:19 GMT", next.publishedDate.toGMTString())
    }

    private fun readTestFile(filename: String): String {
        return IOUtils.toString(
            FileInputStream(javaClass.classLoader.getResource(filename)!!.file),
            StandardCharsets.UTF_8
        )
    }
}
